src/search.cpp

```diff
@@ -1072,7 +1075,8 @@ namespace {
       // Step 16. Reduced depth search (LMR). If the move fails high it will be
       // re-searched at full depth.
       if (    depth >= 3 * ONE_PLY
-          &&  moveCount > 1 + 3 * rootNode
+          &&  moveCount > 1 + 2 * rootNode
+          && (!rootNode || thisThread->bestMoveCount(move) == 0)
```

Comment:
Would it make sense to move this condition last since calling bestMoveCount(move) traverses the move list and so is likely slower than the other conditions?

Reply:
In this case the effect would be insignificant because the  funtion best_move_count is only called at the root. And at non-PV nodes line 1079 wouldn't even be compiled in the code because of template paramter PvNode=false is evaluated to true.

But if we use best_move_count at inner nodes this could be important.